pkg/controller/ephemeralcluster/prowjobreconciler.go

```diff
@@ -72,8 +72,6 @@ func (r *prowJobReconciler) Reconcile(ctx context.Context, req reconcile.Request
 		return reconcile.Result{}, reconcile.TerminalError(fmt.Errorf("%s doesn't have the EC label", pj.Name))
 	}
 
-	pj = pj.DeepCopy()
```

Comment:
why this is removed?

Reply:
The controller runtime internal cache is already making a copy of the object (see [this](https://github.com/kubernetes-sigs/controller-runtime/blob/bbcaf6a9d9ff0ba256e18800f194d387b96dca33/pkg/cache/internal/cache_reader.go#L84-L90)). This wasn't the case in the older versions but we use a recent one now, therefore we can rely on that behavior.